### ISSUE TITLE
ci: Add wheel building job for PRs

### ISSUE THREAD

**Issue by MyPyDavid** [rdmorganiser/rdmo#802]
### Rationale / Begründung

Sometimes, a developer wants to (easily) check out a feature of a branch/PR. 
It requires more effort now that the front- and back-end have been partly separated.

The GH Actions workflow could build a wheels artifact (after tests have passed) which can be used to install a clean version of the RDMO version of that branch/PR.
The same build is already done for the [RDMO releases](https://rdmo.readthedocs.io/en/latest/development/releases.html)

### Affected

developers

### Minimal functionality

* have an `.whl` artifact that can be used to install a rdmo version of the branch.
  * artifact should be build only after the tests have passed
  * naming: `rdmo-<branch>.whl` ?
 

### References / Verweise

* https://rdmo.readthedocs.io/en/latest/development/releases.html
* https://pip.pypa.io/en/stable/user_guide/#installing-from-wheels
* https://github.com/pypa/cibuildwheel
* https://github.com/hynek/build-and-inspect-python-package

**Comment by afuetterer** [comment]
I can work on this issue, if you like.

I propose the name rdmo-<pr-number>.whl (e.g. `rdmo-769.whl`), the branch name can be misleading. Tests are only run on pushes to master and on pull requests. The pr-number identifies a combination of base branch and pull request branch.

You want this wheel to be uploaded to GitHub, only after tests are passing? Your use case is, that you would download the wheel to test a new feature locally, right?

**Comment by afuetterer** [comment]
Can you have at look at the uploaded artifacts of `build-and-inspect-python-package` for pytest here?
https://github.com/pytest-dev/pytest/actions/runs/6699628637

Is this alright for you or overkill?

**Comment by afuetterer** [comment]
The default name of a wheel is rdmo-{version}-py3-none-any.whl, e.g. `rdmo-2.0.1-py3-none-any.whl`.

Should it just stay like that?

What do you prefer?

- [ ] rdmo-{version}-py3-none-any.whl
- [ ] rdmo-{branch-name}.whl
- [ ] rdmo-{pr-number}.whl

I think, it should either stay the default or be rdmo-{pr-number}.whl.
